From b5a6ffc9f56ac78fc1fff29ba10239bc81802e3c Mon Sep 17 00:00:00 2001
From: Alex Deymo <deymo@chromium.org>
Date: Fri, 24 May 2013 12:49:11 -0700
Subject: [PATCH] audio: Disable AVRCP profiles.

The algorithm to guess the role in the AVRCP connection has some bugs
in BlueZ 5.4 that make the org.bluez.Device1.Connect call to hang and
not return on certain circumstances.

This fix disables both AVRCP profiles (remote and target) to avoid
reaching those bugs, given that those profiles are not used anyway.
---
 profiles/audio/manager.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/profiles/audio/manager.c b/profiles/audio/manager.c
index 4517815..d246080 100644
--- a/profiles/audio/manager.c
+++ b/profiles/audio/manager.c
@@ -446,8 +446,6 @@ int audio_manager_init(GKeyFile *conf)
 
 	btd_profile_register(&a2dp_source_profile);
 	btd_profile_register(&a2dp_sink_profile);
-	btd_profile_register(&avrcp_remote_profile);
-	btd_profile_register(&avrcp_target_profile);
 
 	btd_register_adapter_driver(&media_driver);
 
@@ -463,8 +461,6 @@ void audio_manager_exit(void)
 
 	btd_profile_unregister(&a2dp_source_profile);
 	btd_profile_unregister(&a2dp_sink_profile);
-	btd_profile_unregister(&avrcp_remote_profile);
-	btd_profile_unregister(&avrcp_target_profile);
 
 	btd_unregister_adapter_driver(&media_driver);
 }
-- 
1.8.2.1

